cocotb/handle.py

```diff
@@ -530,11 +519,19 @@ def _range_iter(self, left, right):
 
     @NonHierarchyObject.value.getter
     def value(self):
-        # need to iterate over the sub-object
-        result = []
-        for x in range(len(self)):
-            result.append(self[x].value)
-        return result
+        # need to iterate over the sub-objects
+        return [sub_handle.value for sub_handle in self]
```

Comment:
I'm a little worried that actually you do need your original code here, because `__iter__` has a rather scary `except IndexError` that probably isn't what you want.

Reply:
Namely:
```suggestion
        return [
            self[i]
            for i in self._range_iter(self._range[0], self._range[1])
        ]
```